sdk/core/core-rest-pipeline/src/interfaces.ts

```diff
@@ -103,10 +103,20 @@ export interface PipelineRequest {
    */
   headers: HttpHeaders;
 
+  /**
+   * On Node.js, this specifies the number of milliseconds a request can take on stablishing a connection.
+   * If the request is terminated, an `AbortError` is thrown.
+   * Defaults to 0, which disables the timeout.
+   */
+  connectionTimeoutInMs?: number;
+
   /**
    * The number of milliseconds a request can take before automatically being terminated.
    * If the request is terminated, an `AbortError` is thrown.
    * Defaults to 0, which disables the timeout.
+   * On Node.js:
+   * - If `connectionTimeoutInMs` is not set, this timeout could trigger an error if the connection is not established before the timeout finishes.
+   * - If `connectionTimeoutInMs` is set, this timeout will only start counting after the connection is established.
```

Comment:
This seems overly convoluted. `timeout` today means "this request will fail if more than x milliseconds of clock time has elapsed."

I don't think we should change this to be "this request will fail if more than x milliseconds of clock time has elapsed unless you set `connectionTimeoutInMs` to y milliseconds in which case it will only fail after either y milliseconds or x+y milliseconds depending how long the connection time takes"

Reply:
Let’s have a meeting! I’m sure we will be able to find a good approach through a meeting.